src/Shared/FileUtilities.cs

```diff
@@ -550,13 +551,27 @@ internal static string GetFullPathNoThrow(string path)
         /// <returns></returns>
         internal static bool ComparePathsNoThrow(string first, string second, string currentDirectory)
         {
-            // todo: on xplat, figure out if file system is case sensitive or not
-            var stringComparison = StringComparison.OrdinalIgnoreCase;
 
-            var firstFullPath = GetFullPathNoThrow(first, currentDirectory);
-            var secondFullPath = GetFullPathNoThrow(second, currentDirectory);
+            // perf: try comparing the bare strings first
+            if (string.Equals(first, second, PathComparison))
+            {
+                return true;
+            }
+
+            var firstFullPath = NormalizePathForComparisonNoThrow(first, currentDirectory);
+            var secondFullPath = NormalizePathForComparisonNoThrow(second, currentDirectory);
+
+            return string.Equals(firstFullPath, secondFullPath, PathComparison);
+        }
 
-            return string.Equals(firstFullPath, secondFullPath, stringComparison);
+        /// <summary>
+        /// Normalizes a path for path comparison
+        /// Does not throw IO exceptions. See <see cref="GetFullPathNoThrow(string, string)"/>
+        /// 
+        /// </summary>
+        internal static string NormalizePathForComparisonNoThrow(string path, string currentDirectory)
+        {
+            return GetFullPathNoThrow(path, currentDirectory).NormalizeForPathComparison();
```

Comment:
I had to change this from TrimTrailingSlashes to NormalizeForPathComparision because, on xplat, FileUtilities.NormalizePath is now using .Net's GetFullPath method instead of its own implementation. .Net's GetFullPath throws an exception for illegal path args (if there is a `*` in the path) whereas MSBuild's implementation does not.